packages/ui/src/components/pages/SetupWallet/NewWallet/BackupSecretRecoveryPhrase.tsx

```diff
@@ -1,63 +1,25 @@
-import { ChangeEvent, FC, FormEvent, useState } from 'react';
+import { FC } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useDispatch, useSelector } from 'react-redux';
-import { useEffectOnce } from 'react-use';
-import { generateMnemonic } from '@polkadot/util-crypto/mnemonic/bip39';
-import { LoadingButton } from '@mui/lab';
-import { Button, Checkbox, FormControlLabel, FormGroup } from '@mui/material';
 import SecretRecoveryPhrase from 'components/shared/SecretRecoveryPhrase';
-import useSetupWallet from 'hooks/wallet/useSetupWallet';
-import { setupWalletActions } from 'redux/slices/setup-wallet';
-import { RootState } from 'redux/store';
-import { Props, NewWalletScreenStep } from 'types';
-
-const BackupSecretRecoveryPhrase: FC<Props> = ({ className = '' }) => {
+import { Props } from 'types';
+
+interface BackupSecretRecoveryPhraseProps extends Props {
+  secretPhrase: string;
+  title?: string;
+}
+
+const BackupSecretRecoveryPhrase: FC<BackupSecretRecoveryPhraseProps> = ({
+  secretPhrase,
+  title = '',
+  className = '',
+}) => {
   const { t } = useTranslation();
-  const dispatch = useDispatch();
-  const { password } = useSelector((state: RootState) => state.setupWallet);
-  const [checked, setChecked] = useState<boolean>(false);
-  const [secretPhrase, setSecretPhrase] = useState<string>('');
-  const { setup, loading } = useSetupWallet({ secretPhrase, password });
-
-  useEffectOnce(() => {
-    setSecretPhrase(generateMnemonic(12));
-  });
-
-  const doSetupWallet = (e: FormEvent) => {
-    e.preventDefault();
-
-    setup();
-  };
-
-  const back = () => {
-    dispatch(setupWalletActions.setNewWalletScreenStep(NewWalletScreenStep.ChooseWalletPassword));
-  };
-
-  const handleCheckbox = (event: ChangeEvent<HTMLInputElement>) => {
-    setChecked(event.target.checked);
-  };
 
   return (
     <div className={className}>
-      <h3>{t<string>('Finally, back up your secret recovery phrase')}</h3>
+      <h3>{t<string>(title)}</h3>
```

Comment:
Same here, if the title is optional, we should hide the `h3` tag if the title is empty.